code/modules/admin/news.dm

```diff
@@ -0,0 +1,43 @@
+//Handles server news
+#define NEWSFILE "data/news.sav"	//where the memos are saved
+
+/client/
+	var/last_news_hash = null // Stores a hash of the last news window it saw, which gets compared to the current one to see if it is different.
+
+// Returns true if news was updated since last seen.
+/client/proc/check_for_new_server_news()
+	var/savefile/F = get_server_news()
+	if(F)
+		if(md5(F["body"]) != last_news_hash)
+			return TRUE
+	return FALSE
+
+/client/proc/modify_server_news()
+	set name = "Modify Public News"
+	set category = "Server"
+
+	if(!check_rights(0))
+		return
+
+	var/savefile/F = new(NEWSFILE)
+	if(F)
+		var/title = F["title"]
+		var/body = F["body"]
+		var/new_title = copytext(sanitize(input(src,"Write a good title for the news update.  Note: HTML is NOT supported.","Write News", title) as null|text),1,MAX_MESSAGE_LEN)
+		if(!new_title || findtext(new_title,"<script",1,0)) // Is this needed with sanitize()?)
+			return
```

Comment:
Well the button is on the server tab. Anyone with access to that can fuck everything up in worse ways. So I'll just remove it.